File: UVMonitor/Models/UVLevel.swift
import SwiftUI

enum UVLevel: String {
    case low = "Low"
    case moderate = "Moderate"
    case high = "High"
    case veryHigh = "Very High"
    case extreme = "Extreme"

    init(index: Double) {
        switch index {
        case ..<3: self = .low
        case 3..<6: self = .moderate
        case 6..<8: self = .high
        case 8..<11: self = .veryHigh
        default: self = .extreme
        }
    }

    var color: Color {
        switch self {
        case .low: .green
        case .moderate: .yellow
        case .high: .orange
        case .veryHigh: .red
        case .extreme: .purple
        }
    }

    var protectionAdvice: String {
        switch self {
        case .low:
            "No protection needed. Enjoy the outdoors!"
        case .moderate:
            "Wear sunscreen and a hat if outside for extended periods."
        case .high:
            "Sun protection required. Wear sunscreen, hat, and sunglasses."
        case .veryHigh:
            "Extra protection needed. Minimise sun exposure 10am–2pm."
        case .extreme:
            "Avoid being outside during midday hours. Full protection essential."
        }
    }

    var needsProtection: Bool {
        self != .low
    }
}

File: android/app/src/main/java/com/uvmonitor/app/model/UVLevel.kt
package com.uvmonitor.app.model

import androidx.compose.ui.graphics.Color

enum class UVLevel(
    val label: String,
    val color: Color,
    val protectionAdvice: String,
    val needsProtection: Boolean
) {
    LOW(
        label = "Low",
        color = Color(0xFF4CAF50),
        protectionAdvice = "No protection needed. Enjoy the outdoors!",
        needsProtection = false
    ),
    MODERATE(
        label = "Moderate",
        color = Color(0xFFFFC107),
        protectionAdvice = "Wear sunscreen, a hat and sunglasses.",
        needsProtection = true
    ),
    HIGH(
        label = "High",
        color = Color(0xFFFF9800),
        protectionAdvice = "Wear SPF 30+ sunscreen, protective clothing, hat and sunglasses. Seek shade during midday.",
        needsProtection = true
    ),
    VERY_HIGH(
        label = "Very High",
        color = Color(0xFFF44336),
        protectionAdvice = "Minimise sun exposure between 10 am and 2 pm. Apply SPF 50+ sunscreen, wear protective clothing.",
        needsProtection = true
    ),
    EXTREME(
        label = "Extreme",
        color = Color(0xFF9C27B0),
        protectionAdvice = "Avoid being outside during midday hours. Shirt, sunscreen, hat, sunglasses and shade are essential.",
        needsProtection = true
    );

    companion object {
        fun fromUVIndex(index: Double): UVLevel = when {
            index < 3.0 -> LOW
            index < 6.0 -> MODERATE
            index < 8.0 -> HIGH
            index < 11.0 -> VERY_HIGH
            else -> EXTREME
        }
    }
}
